### DISCUSSION TITLE
Promote to Production vs Production Branch

### DISCUSSION THREAD

**Discussion by jsw** [vercel/community#868] [Help] [answered]
I'm new to Vercel and don't see any documentation on the `Promote to Production` functionality. It seems to work as expected and I would prefer to use it over maintaining a production branch, but curious what the tradeoffs  and caveats are. I would like to leave the `Production Branch` field empty, but it defaults to `main`, which I don't want. Looks like I'll have to keep a dummy production branch in my repo and ensure it doesn't trigger a deployment, I believe via `verson.json`. Any tips are appreciated.

**Comment by amyegan** [comment, marked as answer]
Hi @jsw! This sounds like an unusual situation, but I think there's a way to get what you want. You could use an [Ignored Build Step](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel) script to stop production deployments and allow preview deployments. Another option is to stop using the Git integration (set [github.enabled](https://vercel.com/docs/project-configuration#git-configuration/github-enabled) to false in your vercel.json) and [customize GitHub Actions](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) for your Project to only run preview deployments.

**Reply by jsw** [reply]
Hi @amyegan. Are you saying it's unusual to exclusively use  `Promote to Production`? It seems safer for the team to have a single path to production: either promote a deployment or push to the production branch. For reference, my `main` is setup to be a preview deployment, and prior to Vercel we triggered a production deployment via GitHub release from `main`, no production branch. I'm happy to do things more usual, though I like `main` going to dev and button click instead of git branch fu to go to prod. Open to suggestions for working more smoothly with Vercel.

**Reply by amyegan** [reply]
Most people use a `main` or `production` branch and only merge to that branch when they are ready to push the code to production. But that doesn't mean you need to do it that way! There's nothing wrong with preferring a manual Production Deployment step. You can use Ignored Build Step or GitHub Actions to get control of when Production Deployments happen.

**Reply by jsw** [reply]
One disadvantage of using `Promote to Production` is that it's not easy to find the production deployment history, say to revert or for troubleshooting the past. Vercel doesn't show the production deployment history and there's no way to search/filter for production deployments. The only way would be manually sift through all the deployments. I suppose I could add a hook to tag the commit that was promoted so I had the deployment history recorded in git?

**Reply by amyegan** [reply]
[Git tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging) is a good strategy for tracking Production deployments. And you could set up a GitHub Action to redeploy the tagged version manually if you wanted to roll back a problematic deployment.

You can also filter deployments by source branch to make it a little easier to find recent builds 
<img width="522" alt="image" src="https://user-images.githubusercontent.com/8247105/190787306-b1784459-0737-48aa-9154-1a0b176d6c74.png">

**Reply by jsw** [reply]
I see that Vercel creates GitHub deployments, which can be filtered by environment, so I can work with that for now. Thanks for the tips!

<img width="399" alt="Screen Shot 2022-09-17 at 4 45 02 PM" src="https://user-images.githubusercontent.com/768828/190879827-9e9bcd0e-bbd1-4886-87ca-0e91b482b26c.png">
